README.md

```diff
@@ -2,7 +2,217 @@
+* Tree: an Abstract Syntax Tree (AST) or a subtree thereof.
+  In the context of tasty-query, they refer to TASTy trees, which carry their symbol and type information.
+* Type: a type in a Scala program, such as `Int`, `List[String]` or `Bar { type Foo <: Baz[String] }`.
+* Pickle: the public symbol and type information stored in Scala 2 binaries.
+* Elaboration: the process of turning a Scala source program into a typed AST with TASTy-level information.
+  Elaboration includes type inference, implicit resolution, overload resolution and macro expansion.
+  It is the process that gives *semantics* (meaning) to a Scala program.
+  Because of the nature of Scala, it is impossible to only *typecheck* a source program; it must be fully *elaborated*.
+* TASTy: fully elaborated abstract syntax trees for Scala 3 programs.
+
+### Reading TASTy files
+
+TASTy files are complex beasts.
+They represent everything there is to know about the semantics of a Scala 3 program, and then some.
+Whereas Java class files can be read in isolation, and processed to some extent without knowing the classpath context, it is virtually impossible to make any sense of TASTy files without a full classpath.
+TASTy inheritently contains *cycles* between symbols, trees, and types.
+They also leave out some information implicit, to be reconstructed later, such as the type of most expression trees.
+
+Reading TASTy files is therefore complex, and requires a full classpath to make sense of it.
+The product is a multi-dimensional graph involving symbols, trees and types.
+
+tasty-query's job is to read TASTy files for you, and present the information it contains (explicitly or implicitly) in as convenient way as possible.
+
+### Compatibility
+
+In general, we can talk about the compatibility between two "environments" A and B.
+Assuming B comes *after* A (for some definition of "after" such as version-based), then we can define *backward* and *forward* compatibility:
+
+* *Backward* compatibility: if something "works" with environment A, it also works with environment B.
+* *Forward* compatibility: if something works with environment B, it also works with environment A.
+
+By default, and otherwise specified, we refer to *backward* compatibility.
+
+In the Scala 2 ecosystem, there were two kinds of compatibility: *source* and *binary* compatibility.
+
+* Source compatibility: if a program source successfully compiles with environment A, then it also successfully compiles, with the same meaning, with environment B.
+* Binary compatibility: if a program's classfiles successfully link with environment A, then they also successfully link, with the same meaning, with environment B.
+
+Scala 3 adds a third kind of compatibility:
+
+* TASTy compatibility: if a program's TASTy files successfully link with environment A, then they also successfully link, with the same meaning, with environment B.
```

Comment:
I think it would be good to detail a little bit more what this linking step is, because it is not the same as the one used in the definition of binary compatibility (unless I’m wrong?).